lib/block_view.go

```diff
@@ -4766,95 +5644,753 @@ func (bav *UtxoView) _connectUpdateProfile(
+		//
+		// Creator percentage and stake multiple are sufficiently checked above.
+
+		// In this case we need to set all the fields using what was passed
+		// into the transaction.
+
+		// If below block height, user transaction public key.
+		// If above block height, use ProfilePublicKey if available.
+		profileEntryPublicKey := txn.PublicKey
+		if blockHeight > ParamUpdaterProfileUpdateFixBlockHeight {
+			profileEntryPublicKey = profilePublicKey
+		}
+
+		newProfileEntry = ProfileEntry{
+			PublicKey:   profileEntryPublicKey,
+			Username:    txMeta.NewUsername,
+			Description: txMeta.NewDescription,
+			ProfilePic:  txMeta.NewProfilePic,
+
+			CoinEntry: CoinEntry{
+				CreatorBasisPoints: txMeta.NewCreatorBasisPoints,
+
+				// The other coin fields are automatically set to zero, which is an
+				// appropriate default value for all of them.
+			},
+
+			// TODO(DELETEME): This field is deprecated and should be deleted because we're
+			// not allowing staking to profiles.
+			StakeMultipleBasisPoints: txMeta.NewStakeMultipleBasisPoints,
+			// TODO(DELETEME): This field is deprecated and should be deleted because we're
+			// not allowing staking to profiles.
+			StakeEntry: NewStakeEntry(),
+		}
+	}
+	// At this point the newProfileEntry should be set to what we actually
+	// want to store in the db.
+
+	if verifySignatures {
+		// _connectBasicTransfer has already checked that the transaction is
+		// signed by the top-level public key, which we take to be the poster's
+		// public key.
+	}
+
+	// Delete the old profile mappings. Not doing this could cause a username
+	// change to have outdated mappings, among other things.
+	if prevProfileEntry != nil {
+		bav._deleteProfileEntryMappings(prevProfileEntry)
+	}
+
+	// Save the profile entry now that we've updated it or created it from scratch.
+	bav._setProfileEntryMappings(&newProfileEntry)
+
+	// Add an operation to the list at the end indicating we've updated a profile.
+	utxoOpsForTxn = append(utxoOpsForTxn, &UtxoOperation{
+		Type:             OperationTypeUpdateProfile,
+		PrevProfileEntry: prevProfileEntry,
+	})
+
+	return totalInput, totalOutput, utxoOpsForTxn, nil
+}
+
+func (bav *UtxoView) _connectCreateNFT(
+	txn *MsgBitCloutTxn, txHash *BlockHash, blockHeight uint32, verifySignatures bool) (
+	_totalInput uint64, _totalOutput uint64, _utxoOps []*UtxoOperation, _err error) {
+
+	// Check that the transaction has the right TxnType.
+	if txn.TxnMeta.GetTxnType() != TxnTypeCreateNFT {
+		return 0, 0, nil, fmt.Errorf("_connectCreateNFT: called with bad TxnType %s",
+			txn.TxnMeta.GetTxnType().String())
+	}
+	txMeta := txn.TxnMeta.(*CreateNFTMetadata)
+
+	// Validate the txMeta.
+	if txMeta.NumCopies > bav.GlobalParamsEntry.MaxCopiesPerNFT {
+		return 0, 0, nil, RuleErrorTooManyNFTCopies
+	}
+	if txMeta.NumCopies == 0 {
+		return 0, 0, nil, RuleErrorNFTMustHaveNonZeroCopies
+	}
+	royaltyBasisPoints := txMeta.NFTRoyaltyToCreatorBasisPoints + txMeta.NFTRoyaltyToCoinBasisPoints
```

Comment:
OVERFLOW! You need to do overflow checks before you add numbers always:

```
if math.MaxUint64 - txMeta.NFTRoyaltyToCreatorBasisPoints > txMeta.NFTRoyaltyToCreatorBasisPoints {
  error
}
```

Reply:
Added a check and a test for this.  Great catch!
![image](https://user-images.githubusercontent.com/81658108/127058473-99a3958c-dc82-4548-9036-a78c5e9e0568.png)